internal/service/federatedsettingsidentityprovider/model_federated_settings_identity_provider.go

```diff
@@ -0,0 +1,285 @@
+	sort.Sort(mRoleAssignmentV2(roleAssignments))
+
+	var roleAssignmentsMap []map[string]any
+
+	if len(roleAssignments) > 0 {
+		roleAssignmentsMap = make([]map[string]any, len(roleAssignments))
+
+		for i := range roleAssignments {
+			roleAssignmentsMap[i] = map[string]any{
+				"group_id": roleAssignments[i].GroupId,
+				"org_id":   roleAssignments[i].OrgId,
+				"role":     roleAssignments[i].Role,
+			}
+		}
+	}
+
+	return roleAssignmentsMap
+}
+
+func FlattenFederatedUser(federatedUsers []admin.FederatedUser) []map[string]any {
+	var userConflictsMap []map[string]any
+
+	if len(federatedUsers) == 0 {
+		return nil
+	}
+	userConflictsMap = make([]map[string]any, len(federatedUsers))
+
+	for i := range federatedUsers {
+		userConflictsMap[i] = map[string]any{
+			"email_address":          federatedUsers[i].EmailAddress,
+			"federation_settings_id": federatedUsers[i].FederationSettingsId,
+			"first_name":             federatedUsers[i].FirstName,
+			"last_name":              federatedUsers[i].LastName,
+			"user_id":                federatedUsers[i].UserId,
+		}
+	}
+
+	return userConflictsMap
+}
+
+type authFederationoleMappingsByGroupName []admin.AuthFederationRoleMapping
+
+func (ra authFederationoleMappingsByGroupName) Len() int      { return len(ra) }
+func (ra authFederationoleMappingsByGroupName) Swap(i, j int) { ra[i], ra[j] = ra[j], ra[i] }
+
+func (ra authFederationoleMappingsByGroupName) Less(i, j int) bool {
+	return ra[i].ExternalGroupName < ra[j].ExternalGroupName
+}
+
+func FlattenAuthFederationRoleMapping(roleMappings []admin.AuthFederationRoleMapping) []map[string]any {
+	sort.Sort(authFederationoleMappingsByGroupName(roleMappings))
+
+	var roleMappingsMap []map[string]any
+
+	if len(roleMappings) > 0 {
+		roleMappingsMap = make([]map[string]any, len(roleMappings))
+
+		for i := range roleMappings {
+			roleMappingsMap[i] = map[string]any{
+				"external_group_name": roleMappings[i].ExternalGroupName,
+				"id":                  roleMappings[i].Id,
+				"role_assignments":    FlattenRoleAssignmentsV2(roleMappings[i].RoleAssignments),
+			}
+		}
+	}
+
+	return roleMappingsMap
+}
+
+func FlattenPemFileInfo(pemFileInfo admin.PemFileInfo) []map[string]any {
+	var pemFileInfoMap []map[string]any
+
+	if len(pemFileInfo.Certificates) > 0 {
+		pemFileInfoMap = make([]map[string]any, 1)
+
+		pemFileInfoMap[0] = map[string]any{
+			"certificates": FlattenFederatedSettingsCertificates(pemFileInfo.Certificates),
+			"file_name":    pemFileInfo.FileName,
+		}
+	}
+
+	return pemFileInfoMap
+}
+
+func FlattenFederatedSettingsCertificates(certificates []admin.X509Certificate) []map[string]any {
+	var certificatesMap []map[string]any
+
+	if len(certificates) > 0 {
+		certificatesMap = make([]map[string]any, len(certificates))
+
+		for i := range certificates {
+			certificatesMap[i] = map[string]any{
+				"not_after":  conversion.TimeToString(*certificates[i].NotAfter),
+				"not_before": conversion.TimeToString(*certificates[i].NotBefore),
```

Comment:
`TimePtrToStringPtr` can be used here to avoid accessing pointer value directly

Reply:
Thanks for the suggestion. Changed